source/write/transactions.txt

```diff
@@ -0,0 +1,133 @@
+multiple sessions and transactions instead of instantiating a new client each
+time.
+
+.. warning::
+
+   Use a ``ClientSession`` only with the ``MongoClient`` (or associated
+   ``MongoDatabase`` or ``MongoCollection``) that created it. Using a ``ClientSession`` with
+   a different ``MongoClient`` results in operation errors.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` and ``sample_mflix.movies`` collections
+from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:ref:`<java-rs-getting-started>`.
+
+.. include:: includes/reactor-note.rst
+
+Transaction Methods
+-------------------
+
+Create a ``ClientSession`` by using the ``startSession()`` method on your ``MongoClient``
+instance. You can then modify the session state by using the methods provided by
+the ``ClientSession``. The following table details the methods you can use to
+manage your transaction:
+
+.. list-table::
+   :widths: 40 60
+   :header-rows: 1
+ 
+   * - Method
+     - Description
+ 
+   * - ``startTransaction()``
+     - | Starts a new transaction, configured with the given options, on
+         this session. Throws an exception if there is already
+         a transaction in progress for the session. To learn more about
+         this method, see the :manual:`startTransaction() page
+         </reference/method/Session.startTransaction/>`  in the {+mdb-server+} manual.
+ 
+   * - ``abortTransaction()``
+     - | Ends the active transaction for this session. Throws an exception
+         if there is no active transaction for the session or if the
+         transaction is already committed or ended. To learn more about
+         this method, see the :manual:`abortTransaction() page
+         </reference/method/Session.abortTransaction/>`  in the {+mdb-server+} manual.
+ 
+   * - ``commitTransaction()``
+     - | Commits the active transaction for this session. Throws an exception
+         if there is no active transaction for the session or if the
+         transaction was ended. To learn more about
+         this method, see the :manual:`commitTransaction() page
+         </reference/method/Session.commitTransaction/>`  in the {+mdb-server+} manual.
+
+Transaction Example
+-------------------
+
+The following example demonstrates how to create a session, create a
+transaction, and insert documents into multiple collections in one transaction.
+The code completes the following steps:
+
+- Creates a session from the client by using the ``startSession()`` method
+- Starts a transaction by using the ``startTransaction()`` method
+- Inserts documents into the ``restaurants`` and ``movies`` collections
+- Commits the transaction by using the ``commitTransaction()`` method
+
+.. literalinclude:: /includes/write/transactions/transactions.java
+   :language: java
+   :copyable:
+   :emphasize-lines: 26, 29, 33-36, 38
```

Comment:
Suggestion: Trim some of the code sample – I think the most helpful part to include is everything contained within the `try` block.